test/unit/math/fwd/mat/vectorize/expect_val_deriv_eq.hpp

```diff
@@ -3,8 +3,11 @@
 
 #include <stan/math/fwd/mat.hpp>
 #include <gtest/gtest.h>
+#include <stan/math/prim/scal/fun/is_nan.hpp>
 
 static inline void expect_val_deriv_eq(double exp_var, double test_var) {
+  using stan::math::is_nan;
+  if (is_nan(exp_var) && is_nan(test_var)) return;
```

Comment:
NaN doesn't fit really well within the testing framework that I've created:
1) It's not an error so any tests using EXPECT_THROW will fail.
2) A lot of cmath functions will return NaN instead of throwing errors.
3) There are functions like cbrt, which are defined for a value, but their derivatives aren't. 

So, the easiest way to deal with it is to check if both the expected values and test values are NaN. If they are, then it's at least consistent. If they aren't, then there's a problem.